client/driver/executor/checks.go

```diff
@@ -0,0 +1,174 @@
+package executor
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/armon/circbuf"
+	docker "github.com/fsouza/go-dockerclient"
+
+	cstructs "github.com/hashicorp/nomad/client/driver/structs"
+)
+
+var (
+	// We store the client globally to cache the connection to the docker daemon.
+	createClient sync.Once
+	client       *docker.Client
+)
+
+// DockerScriptCheck runs nagios compatible scripts in a docker container and
+// provides the check result
+type DockerScriptCheck struct {
+	id          string
+	interval    time.Duration
+	containerID string
+	logger      *log.Logger
+	cmd         string
+	args        []string
+
+	dockerEndpoint string
+	tlsCert        string
+	tlsCa          string
+	tlsKey         string
+}
+
+// dockerClient creates the client to interact with the docker daemon
+func (d *DockerScriptCheck) dockerClient() (*docker.Client, error) {
+	if client != nil {
+		return client, nil
+	}
+
+	var err error
+	createClient.Do(func() {
+		if d.dockerEndpoint != "" {
+			if d.tlsCert+d.tlsKey+d.tlsCa != "" {
+				d.logger.Printf("[DEBUG] executor.checks: using TLS client connection to %s", d.dockerEndpoint)
```

Comment:
Just wondering if it would be easy to share some code here with the docker driver so as we make improvements to creating the client it doesn't have to be copied in two places